File: src/main/java/API/demo/Rulesets/Ruleset.java
package API.demo.Rulesets;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Ruleset {
    private Integer id;
    private String name;
    private Long creation_date;
    private List<?> rules;
}

File: src/app/Ruleset.ts
export interface Ruleset {
  id: number;
  name: string;
  creation_date: number | string;
  rules: { 
    id: number; 
    priority: number; 
    event_type: string; 
    conditions: {
      id: number, 
      fact_type: string, 
      value_type: string
    }[] 
  }[];
}
